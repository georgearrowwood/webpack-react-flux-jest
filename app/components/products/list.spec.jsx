import React from 'react';
import { shallow } from 'enzyme';

import productsStore from './store';
import productsActions from './actions';
import ProductsList from './list';

const mockProductsList = [
  { id: 1, title: 'test product 1' },
  { id: 2, title: 'test product 2' },
];

beforeAll(() => {
  jest.spyOn(productsActions, 'fetchList').mockImplementation(jest.fn);
});

describe('ProductsList Container', () => {
  it('Renders component and triggers action fetchList on did mount', () => {
    const wrapper = shallow(<ProductsList />);
    expect(wrapper).toMatchSnapshot();
    expect(Array.isArray(wrapper.prop('products'))).toEqual(true);
    expect(wrapper.prop('products').length).toEqual(0);
    expect(productsActions.fetchList.mock.calls.length).toEqual(1);
    productsActions.fetchList.mockReset();
  });

  it('Renders and triggers change manually, loading products from store', () => {
    jest.spyOn(productsStore, 'getList').mockImplementation(() => mockProductsList);
    const wrapper = shallow(<ProductsList />);
    expect(wrapper).toMatchSnapshot();
    expect(Array.isArray(wrapper.prop('products'))).toEqual(true);
    expect(wrapper.prop('products').length).toEqual(0);
    wrapper.instance().onChange();
    expect(productsStore.getList.mock.calls.length).toEqual(1);
    wrapper.update();
    expect(wrapper.prop('products').length).toEqual(2);
    productsStore.getList.mockRestore();
    productsActions.fetchList.mockReset();
  });
});

afterAll(() => {
  productsActions.fetchList.mockRestore();
});
