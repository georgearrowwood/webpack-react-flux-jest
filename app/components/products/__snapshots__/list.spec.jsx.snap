// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList Container Renders and triggers change manually, loading products from store 1`] = `
<Products
  deleteProductHandler={[Function]}
  products={Array []}
/>
`;

exports[`ProductsList Container Renders component and triggers action fetchList on did mount 1`] = `
<Products
  deleteProductHandler={[Function]}
  products={Array []}
/>
`;
